lib/model-interfaces/langchain/functions/request-handler/adapters/bedrock/base.py

```diff
@@ -1,97 +1,128 @@
+        logger.debug(f"ChatPromptTemplate messages: {chat_prompt_template.messages}")
+
+        return chat_prompt_template
+
     def get_prompt(self):
-        prompt_template = ChatPromptTemplate(
+        # Fetch the conversation prompt based on the current language
+        conversation_prompt = prompts[locale]["conversation_prompt"]
+        logger.info("Generating general conversation prompt template.")
+        chat_prompt_template = ChatPromptTemplate.from_messages(
             [
-                (
-                    "system",
-                    (
-                        "The following is a friendly conversation between "
-                        "a human and an AI."
-                        "If the AI does not know the answer to a question, it "
-                        "truthfully says it does not know."
-                    ),
-                ),
+                ("system", conversation_prompt),
                 MessagesPlaceholder(variable_name="chat_history"),
                 ("human", "{input}"),
             ]
         )
-
-        return prompt_template
+        # Trace the ChatPromptTemplate by logging its content
+        logger.debug(f"ChatPromptTemplate messages: {chat_prompt_template.messages}")
+        return chat_prompt_template
 
     def get_condense_question_prompt(self):
-        contextualize_q_system_prompt = (
-            "Given the following conversation and a follow up"
-            " question, rephrase the follow up question to be a standalone question."
-        )
-        return ChatPromptTemplate.from_messages(
+        # Fetch the prompt based on the current language
+        condense_question_prompt = prompts[locale]["condense_question_prompt"]
+        logger.info("Generating condense question prompt template.")
+        chat_prompt_template = ChatPromptTemplate.from_messages(
             [
-                ("system", contextualize_q_system_prompt),
+                ("system", condense_question_prompt),
                 MessagesPlaceholder("chat_history"),
                 ("human", "{input}"),
             ]
         )
+        # Trace the ChatPromptTemplate by logging its content
+        logger.debug(f"ChatPromptTemplate messages: {chat_prompt_template.messages}")
+        return chat_prompt_template
 
     def get_llm(self, model_kwargs={}, extra={}):
         bedrock = genai_core.clients.get_bedrock_client()
         params = {}
-        if "temperature" in model_kwargs:
-            params["temperature"] = model_kwargs["temperature"]
-        if "topP" in model_kwargs:
-            params["top_p"] = model_kwargs["topP"]
-        if "maxTokens" in model_kwargs:
-            params["max_tokens"] = model_kwargs["maxTokens"]
 
+        # Collect temperature, topP, and maxTokens if available
+        temperature = model_kwargs.get("temperature")
+        top_p = model_kwargs.get("topP")
+        max_tokens = model_kwargs.get("maxTokens")
+
+        if temperature:
```

Comment:
It would not set the value if temperature is 0
```suggestion
        if temperature is not None:
```